rpc/src/v1/impls/transactions_pool.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::{Arc, mpsc};
+use std::thread;
+
+use jsonrpc_core::Result;
+use jsonrpc_core::futures::Future;
+use jsonrpc_pubsub::{SubscriptionId, typed::{Sink, Subscriber}};
+
+use v1::helpers::{errors, Subscribers};
+use v1::metadata::Metadata;
+use v1::traits::TransactionsPool;
+use v1::types::pubsub;
+
+use parity_runtime::Executor;
+use parking_lot::RwLock;
+use ethereum_types::H256;
+
+type Client = Sink<pubsub::Result>;
+
+/// Transactions pool PubSub implementation.
+pub struct TransactionsPoolClient {
+    handler: Arc<TransactionsNotificationHandler>,
+    transactions_pool_subscribers: Arc<RwLock<Subscribers<Client>>>,
+}
+
+impl TransactionsPoolClient {
+    /// Creates new `TransactionsPoolClient`.
+    pub fn new(executor: Executor) -> Self {
+        let transactions_pool_subscribers = Arc::new(RwLock::new(Subscribers::default()));
+        let handler = Arc::new(
+            TransactionsNotificationHandler::new(
+                executor,
+                transactions_pool_subscribers.clone(),
+                )
+        );
+
+        TransactionsPoolClient {
+            handler,
+            transactions_pool_subscribers,
+        }
+    }
+    pub fn run(&self, pool_receiver: mpsc::Receiver<(H256, String)>) {
+        let handler = self.handler.clone();
+        thread::spawn(move || loop {
+            let res = pool_receiver.recv();
+            if let Ok(res) = res {
+                handler.notify_transaction(res)
+            }
+        });
+    }
+
+}
+
+pub struct TransactionsNotificationHandler {
+    executor: Executor,
+    transactions_pool_subscribers: Arc<RwLock<Subscribers<Client>>>,
+}
+
+impl TransactionsNotificationHandler {
+    fn new(executor: Executor, transactions_pool_subscribers: Arc<RwLock<Subscribers<Client>>>) -> Self {
+        TransactionsNotificationHandler {
+            executor,
+            transactions_pool_subscribers,
+        }
+    }
+
+    fn notify(executor: &Executor, subscriber: &Client, result: pubsub::Result) {
+        executor.spawn(subscriber
+            .notify(Ok(result))
+            .map(|_| ())
+            .map_err(|e| warn!(target: "rpc", "Unable to send notification: {}", e))
+        );
+    }
+
+    pub fn notify_transaction(&self, status: (H256, String)) {
+        for subscriber in self.transactions_pool_subscribers.read().values() {
+            let status = (status.0.clone(), status.1.clone());
+            Self::notify(&self.executor, subscriber, pubsub::Result::TransactionStatus(status));
+        }
+    }
+}
+
+impl TransactionsPool for TransactionsPoolClient {
+    type Metadata = Metadata;
+
+    fn subscribe(&self, _meta: Metadata, subscriber: Subscriber<pubsub::Result>, params: Option<pubsub::Params>) {
```

Comment:
using `pubsub::Result` here kind of suggests that we might get different values of this enum as well. I think it should be safe to just use the tuple directly:
```suggestion
    fn subscribe(&self, _meta: Metadata, subscriber: Subscriber<(H256, Status)>, params: Option<pubsub::Params>) {
```